pkg/collector/edp/collector.go

```diff
@@ -2,69 +2,204 @@ package edp
 
 import (
 	"context"
-	"maps"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
 	"time"
 
 	"go.opentelemetry.io/otel"
 	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
 	"go.opentelemetry.io/otel/trace"
-	"go.uber.org/zap"
 
 	"github.com/kyma-project/kyma-metrics-collector/pkg/collector"
 	"github.com/kyma-project/kyma-metrics-collector/pkg/resource"
 	"github.com/kyma-project/kyma-metrics-collector/pkg/runtime"
-	skrcommons "github.com/kyma-project/kyma-metrics-collector/pkg/skr/commons"
 )
 
 type Collector struct {
-	scanners []resource.Scanner
-	logger   *zap.Logger
+	EDPClient *Client
+	scanners  []resource.Scanner
 }
 
 var _ collector.CollectorSender = &Collector{}
 
-func NewCollector(scanner ...resource.Scanner) collector.CollectorSender {
+func NewCollector(EDPClient *Client, scanner ...resource.Scanner) collector.CollectorSender {
 	return &Collector{
-		scanners: scanner,
+		EDPClient: EDPClient,
+		scanners:  scanner,
 	}
 }
 
 func (c *Collector) CollectAndSend(ctx context.Context, runtime *runtime.Info, previousScans collector.ScanMap) (collector.ScanMap, error) {
-	childCtx, span := otel.Tracer("").Start(ctx, "collect",
+	var errs []error
+
+	childCtx, span := otel.Tracer("").Start(ctx, "kmc.collect_scans_and_send_measurements",
 		trace.WithAttributes(
+			attribute.String("instance_id", runtime.InstanceID),
+			attribute.String("runtime_id", runtime.RuntimeID),
+			attribute.String("sub_account_id", runtime.SubAccountID),
+			attribute.String("global_account_id", runtime.GlobalAccountID),
+			attribute.String("shoot_name", runtime.ShootName),
 			attribute.String("provider", runtime.ProviderType),
-			attribute.String("shoot_id", runtime.ShootInfo.ShootName),
 		),
 	)
 	defer span.End()
 
+	currentTimestamp := getTimestampNow()
+
+	scans, err := c.executeScans(childCtx, runtime)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("failed to execute all scans: %w", err))
+		span.RecordError(err)
+		span.SetStatus(codes.Error, err.Error())
+	}
+
+	EDPMeasurements, err := c.convertScansToEDPMeasurements(scans, previousScans, runtime)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("failed to convert all scans to EDP measurements: %w", err))
+		span.RecordError(err)
+		span.SetStatus(codes.Error, err.Error())
+	}
+
+	if len(EDPMeasurements) == 0 {
+		errMsg := "no EDP measurements to send to EDP"
+		errs = append(errs, fmt.Errorf("%s", errMsg))
```

Comment:
lets extract an error for this
errNoMeasurementsSent := errors.New("no measurements sent to EDP")